ui/job-view/context/SelectedJob.jsx

```diff
@@ -0,0 +1,270 @@
+      selectedJob: null,
+    };
+    this.value = {
+      ...this.state,
+      setSelectedJob: this.setSelectedJob,
+      clearSelectedJob: this.clearSelectedJob,
+      changeSelectedJob: this.changeSelectedJob,
+    };
+  }
+
+  componentDidMount() {
+    this.setSelectedJob = this.setSelectedJob.bind(this);
+    this.clearSelectedJob = this.clearSelectedJob.bind(this);
+    this.changeSelectedJob = this.changeSelectedJob.bind(this);
+    this.noMoreUnclassifiedFailures = this.noMoreUnclassifiedFailures.bind(this);
+
+    // TODO: this.value needs to now get the bound versions of the functions.
+    // But when we move the function binding to the constructors, we won't
+    // have to re-do this in componentDidMount.
+    this.value = {
+      ...this.state,
+      setSelectedJob: this.setSelectedJob,
+      clearSelectedJob: this.clearSelectedJob,
+      changeSelectedJob: this.changeSelectedJob,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { jobsLoaded } = this.props;
+
+    if (jobsLoaded !== prevProps.jobsLoaded) {
+      this.setSelectedJobFromQueryString();
+    }
+  }
+
+  setValue(newState, callback) {
+    this.value = { ...this.value, ...newState };
+    this.setState(newState, callback);
+  }
+
+  /**
+   * If the URL has a query string param of ``selectedJob`` then select
+   * that job on load.
+   *
+   * If that job isn't in any of the loaded pushes, then throw
+   * an error and provide a link to load it with the right push.
+   */
+  setSelectedJobFromQueryString() {
+    const { notify } = this.props;
+    const { repoName, selectedJob } = this.state;
+    const selectedJobIdStr = getUrlParam('selectedJob');
+    const selectedJobId = parseInt(selectedJobIdStr);
+
+    if (selectedJobIdStr && (!selectedJob || selectedJob.id !== selectedJobId)) {
+      const jobMap = this.ThResultSetStore.getJobMap();
+      const selectedJobEl = jobMap[selectedJobIdStr];
+
+      // select the job in question
+      if (selectedJobEl) {
+        this.setSelectedJob(selectedJobEl.job_obj);
+      } else {
+        // If the ``selectedJob`` was not mapped, then we need to notify
+        // the user it's not in the range of the current result set list.
+        JobModel.get(repoName, selectedJobId).then((job) => {
+          PushModel.get(job.result_set_id).then(async (resp) => {
+            if (resp.ok) {
+              const push = await resp.json();
+              const newPushUrl = getJobsUrl({ repo: repoName, revision: push.data.revision, selectedJob: selectedJobId });
+
+              this.clearSelectedJob();
+
+              // the job exists, but isn't in any loaded push.
+              // provide a message and link to load the right push
+              notify.send(
+                `Selected job id: ${selectedJobId} not within current push range.`,
+                'danger',
+                { sticky: true, linkText: 'Load push', newPushUrl });
+            }
+          });
+        }).catch((error) => {
```

Comment:
I think you need to throw in error in the previous block otherwise it won't catch a 404 error, only network errors?